src/WSRequestHandler_Sources.cpp

```diff
@@ -76,17 +76,48 @@ RpcResponse WSRequestHandler::CreateSource(const RpcRequest& request)
 	if (request.hasField("setVisible")) {
 		data.setVisible = obs_data_get_bool(request.parameters(), "setVisible");
 	}
-	
+
 	obs_enter_graphics();
 	obs_scene_atomic_update(scene, Utils::AddSourceHelper, &data);
 	obs_leave_graphics();
-	
+
 	OBSDataAutoRelease responseData = obs_data_create();
 	obs_data_set_int(responseData, "itemId", obs_sceneitem_get_id(data.sceneItem));
 
 	return request.success(responseData);
 }
 
+/**
+* Remove the specified source.
+*
+* @param {String} `sourceName` Source name.
+*
+* @api requests
+* @name RemoveSource
+* @category sources
+* @since unreleased
+*/
+RpcResponse WSRequestHandler::RemoveSource(const RpcRequest& request)
+{
+	if (!request.hasField("sourceName")) {
+		return request.failed("missing request parameters");
+	}
+
+	QString sourceName = obs_data_get_string(request.parameters(), "sourceName");
+	if (sourceName.isEmpty()) {
+		return request.failed("invalid request parameters");
+	}
+
+	OBSSourceAutoRelease source = obs_get_source_by_name(sourceName.toUtf8());
+	if (!source) {
+		return request.failed("specified source doesn't exist");
+	}
+
+	obs_source_remove(source);
```

Comment:
This call works, but if the source is referenced in one or more scenes, remove the source directly doesn't trigger a UI update of the sources list in OBS.

For input sources, what do you think of checking first if it is used somewhere, and if so return an error message?

Reply:
It would be nice to have a `force` flag then which would allow to remove it despite the side effects.

But as this would cause a weird state in the UI... Maybe better not to.